python/dataproc_templates/jdbc/jdbc_to_jdbc.py

```diff
@@ -0,0 +1,201 @@
+            ),
+            choices=[
+                constants.OUTPUT_MODE_OVERWRITE,
+                constants.OUTPUT_MODE_APPEND,
+                constants.OUTPUT_MODE_IGNORE,
+                constants.OUTPUT_MODE_ERRORIFEXISTS
+            ]
+        )
+        parser.add_argument(
+            f'--{constants.JDBCTOJDBC_OUTPUT_BATCH_SIZE}',
+            dest=constants.JDBCTOJDBC_OUTPUT_BATCH_SIZE,
+            required=False,
+            default="1000",
+            help='JDBC output batch size. Default set to 1000'
+        )
+
+
+        known_args: argparse.Namespace
+        known_args, _ = parser.parse_known_args(args)
+
+        return vars(known_args)
+
+    def run(self, spark: SparkSession, args: Dict[str, Any]) -> None:
+
+        logger: Logger = self.get_logger(spark=spark)
+
+        # Arguments
+        input_jdbc_url: str = args[constants.JDBCTOJDBC_INPUT_URL]
+        input_jdbc_driver: str = args[constants.JDBCTOJDBC_INPUT_DRIVER]
+        input_jdbc_table: str = args[constants.JDBCTOJDBC_INPUT_TABLE]
+        input_jdbc_partitioncolumn: str = args[constants.JDBCTOJDBC_INPUT_PARTITIONCOLUMN]
+        input_jdbc_lowerbound: str = args[constants.JDBCTOJDBC_INPUT_LOWERBOUND]
+        input_jdbc_upperbound: str = args[constants.JDBCTOJDBC_INPUT_UPPERBOUND]
+        jdbc_numpartitions: str = args[constants.JDBCTOJDBC_NUMPARTITIONS]
+        output_jdbc_url: str = args[constants.JDBCTOJDBC_OUTPUT_URL]
+        output_jdbc_driver: str = args[constants.JDBCTOJDBC_OUTPUT_DRIVER]
+        output_jdbc_table: str = args[constants.JDBCTOJDBC_OUTPUT_TABLE]
+        output_jdbc_create_table_option: str = args[constants.JDBCTOJDBC_OUTPUT_CREATE_TABLE_OPTION]
+        output_jdbc_mode: str = args[constants.JDBCTOJDBC_OUTPUT_MODE]
+        output_jdbc_batch_size: int = args[constants.JDBCTOJDBC_OUTPUT_BATCH_SIZE]
+
+
+        logger.info(
+            "Starting JDBC to JDBC spark job with parameters:\n"
+            f"{pprint.pformat(args)}"
+        )
+        
+        # Read
+        input_data: DataFrame
+        
+        partition_parameters=str(input_jdbc_partitioncolumn) + str(input_jdbc_lowerbound) + str(input_jdbc_upperbound)
+        if ((partition_parameters != "") & ((input_jdbc_partitioncolumn == "") | (input_jdbc_lowerbound == "") | (input_jdbc_upperbound == ""))):
+            logger.error("Set all the sql partitioning parameters together-jdbctojdbc.input.partitioncolumn,jdbctojdbc.input.lowerbound,jdbctojdbc.input.upperbound. Refer to README.md for more instructions.")
+            exit (1)
+        elif (partition_parameters == ""):
+            input_data=spark.read \
+                .format(constants.FORMAT_JDBC) \
+                .option(constants.JDBC_URL, input_jdbc_url) \
+                .option(constants.JDBC_DRIVER, input_jdbc_driver) \
+                .option(constants.JDBC_TABLE, input_jdbc_table) \
+                .option(constants.JDBC_NUMPARTITIONS, jdbc_numpartitions) \
```

Comment:
Please verify if this block works, if numpartition > 1
Even if it does, what is the purpose of specifying partitions number without rest of partition parameters?

Reply:
@shashank-google  yes it works even without the other parameters.  this parameter is used for parallelism in table reading and writing(concurrent JDBC connections). Other parameters are used while reading only, but numpartitions are used for both read and write for parallelism. 

I reverified from the documentation and included these parameters. 
https://spark.apache.org/docs/latest/sql-data-sources-jdbc.html